### DISCUSSION TITLE
A per transaction file trigger that runs before %posttrans scriptlets

### DISCUSSION THREAD

**Discussion by daandemeyer** [rpm-software-management/rpm#3348] [Ideas]
We're looking to move systemd unit operations in rpm scriptlets to `%posttrans` instead of `%postun`. While looking into this, I discovered that currently `systemctl daemon-reload` is run via `%transfiletriggerpostun` which is executed **after** `%posttrans` scriptlets. I'd like to have a transaction scriptlet that is executed before `%posttrans` scriptlets.

Alternatively, having the file trigger priorities apply to `%transfiletriggerpostun` as well so that triggers with a sufficiently high priority are executed before `%posttrans` scriptlets would also solve my issue.

**Comment by dmnks** [comment]
Right, this sounds like a reasonable request. Currently, transactional scriptlets, whether trigger-based or normal, are executed in a fixed order, which means the priority value is not respected for these, as you've noted. I think this deserves an RFE. That's not to say this would be easy to implement, but it does make sense to me.

**Reply by dmnks** [reply]
In fact, there was a [fix](https://github.com/rpm-software-management/rpm/pull/3267/commits/0bcdc100a34303601232ca9c77b2763dc886f348) recently that involved priorities & `%transfiletriggerpostun` but it doesn't address this particular issue.

**Reply by dmnks** [reply]
OK, talked to the team and decided to file a ticket after all: https://github.com/rpm-software-management/rpm/issues/3353

**Reply by daandemeyer** [reply]
Thanks!
